crates/pallets/starknet/src/tests/invoke_tx.rs

```diff
@@ -525,3 +533,144 @@ fn test_verify_nonce_in_unsigned_tx() {
         );
     });
 }
+
+#[test]
+fn test_tx_fail_on_invalid_nonce_should_revert_storage() {
+    new_test_ext::<MockRuntime>().execute_with(|| {
+        basic_test_setup(2);
+
+        let transaction = get_invoke_dummy(Felt252Wrapper::ZERO);
+
+        let tx_sender = transaction.sender_address.into();
+        let tx_source = TransactionSource::InBlock;
+        let call = Call::invoke { transaction: transaction.into() };
+
+        assert!(Starknet::validate_unsigned(tx_source, &call).is_ok());
+
+        set_nonce::<MockRuntime>(&tx_sender, &Nonce(StarkFelt::from(1u64)));
+
+        assert_eq!(
+            Starknet::validate_unsigned(tx_source, &call),
+            Err(TransactionValidityError::Invalid(InvalidTransaction::Stale))
+        );
+
+        let storage_key = (tx_sender, StorageKey(PatriciaKey(StarkFelt::from(0u128))));
+
+        assert_eq!(Starknet::storage(storage_key), StarkFelt::from(0u128));
```

Comment:
I suppose you probably want to check Starknet::nonces here?

Reply:
if yes, then the test case should be `test_tx_fail_on_invalid_nonce_should_not_increment_nonce`

Reply:
hey this was a dummy test - I had assumed that even if transaction fails due to invalid nonce the storage should get reverted. But in practice that is not the case, in case of invalid nonce the transaction gets `failed` and not `reverted`

This test is not needed so removing it